server/src/main/java/com/flipkart/varadhi/db/ZKPersistence.java

```diff
@@ -73,6 +76,27 @@ public boolean exists(String resourceKey) {
         }
     }
 
+    public void delete(String resourceKey) {
+        String resourcePath = getResourcePath(resourceKey);
+        try {
+            zkCurator.delete().forPath(resourcePath);
+        } catch (Exception e) {
+            log.error(String.format("Failed to delete entity %s ", resourcePath), e);
+            throw new VaradhiException(e);
+        }
+    }
+
+    public List<String> list(String resourceKey) {
+        String resourcePath = getResourcePath(resourceKey);
+        try {
+            if (!exists(resourceKey)) return Collections.emptyList();
```

Comment:
is this alright to return empty here? Shouldn't we throw error? Considering that caller wanted to know the children under a path, but the path does not even exist.